odoo/addons/base/views/ir_mail_server_views.xml

```diff
@@ -5,30 +5,37 @@
                 <form string="Outgoing Mail Servers">
-                  <sheet>
-                    <group col="4">
-                        <field name="name"/>
-                        <field name="from_filter"/>
-                        <field name="sequence"/>
-                        <field name="active" widget="boolean_toggle"/>
-                    </group>
-                    <group col="4" string="Connection Information">
-                        <field name="smtp_host"/>
-                        <field name="smtp_port" options="{'format': false}"/>
-                        <field name="smtp_debug" groups="base.group_no_one"/>
-                     </group>
-                     <group>
-                        <group string="Security and Authentication">
-                            <field name="smtp_authentication"/>
-                            <field name="smtp_encryption"/>
-                            <field name="smtp_user" attrs="{'invisible': [('smtp_authentication', '!=', 'login')]}" force_save="1"/>
-                            <field name="smtp_pass" attrs="{'invisible': [('smtp_authentication', '!=', 'login')]}" password="True" force_save="1"/>
-                            <field name="smtp_ssl_certificate" attrs="{'invisible': [('smtp_authentication', '!=', 'certificate')]}" force_save="1"/>
-                            <field name="smtp_ssl_private_key" attrs="{'invisible': [('smtp_authentication', '!=', 'certificate')]}" force_save="1"/>
-                            <button name="test_smtp_connection" type="object" string="Test Connection" icon="fa-television"/>
+                    <header>
+                        <button name="test_smtp_connection" type="object"
+                            class="btn-primary"
+                            string="Test Connection"/>
+                    </header>
+                    <sheet>
+                        <group col="4">
+                            <field name="name" placeholder="e.g. My Outgoing Server"/>
+                            <field name="from_filter"/>
+                            <field name="sequence"/>
+                            <field name="active" widget="boolean_toggle"/>
                         </group>
-                    </group>
-                  </sheet>
+                        <group col="4" string="Connection Information">
+                            <field name="smtp_host"/>
+                            <field name="smtp_port" options="{'format': false}"/>
+                            <field name="smtp_debug" groups="base.group_no_one"/>
+                        </group>
+                        <group>
+                            <group string="Security">
+                                <field name="smtp_authentication" widget="radio"/>
+                                <field name="smtp_encryption" widget="radio"/>
+                            </group>
+                            <group string="Authentication"
```

Comment:
Could have two groups, one for "login" and one for "certificate" to avoid repeating the invisible conditions.
But they are both "Authentication" so I guess you can leave it like that.

Reply:
Indeed invisible conditions are a bit verbose. We could imagine having two groups called "Authentication" and hiding one or another, but not sure it would not flicker and basically, they belong to the same group. I suggest to keep like this.